src/output-json-http.c

```diff
@@ -368,6 +372,46 @@ static void JsonHttpLogJSONExtended(json_t *js, htp_tx_t *tx)
     json_object_set_new(js, "length", json_integer(tx->response_message_len));
 }
 
+static void JsonHttpLogJSONHeaders(json_t *js, uint32_t direction, htp_tx_t *tx)
```

Comment:
I think this function would be cleaner and more compact if the vars would be declared on first use, and in reduced scope. I've made a few specific comments on it, but it applies to most.